includes/gateways/stripe/includes/give-stripe-helpers.php

```diff
@@ -1035,11 +1035,12 @@ function give_stripe_cents_to_dollars( $cents ) {
  * @param string $dollars Amount in dollars.
  *
  * @since  2.5.0
+ * @since 2.9.2  Return amount in cent only if currency is not zero-decimal currency.
  *
  * @return string
  */
 function give_stripe_dollars_to_cents( $dollars ) {
-	return round( $dollars, give_currency_decimal_filter() ) * 100;
+	return give_is_zero_based_currency() ? round( $dollars ) : round( $dollars, give_get_price_decimals() ) * 100;
```

Comment:
@ravinderk Looks like we're still making he assumption here that the way to convert _any_ currency to its cents is to multiply it by 100. Can we instead make use of the new `Money` class to convert it to the base amount based on the currency?

Reply:
@JasonTheAdams Good catch.

fixed: https://github.com/impress-org/givewp/compare/bc8a61cdd512...b85e52d94f5d

Reply:
Minor type in function, I will fix it

Reply:
Fixed